### PR TITLE
addEdge(), removeEdge(), and documentation added to ControlNet

### PR DESCRIPTION
**Pull request by krlberry**

### CODE REVIEW

File: isis/src/control/objs/ControlNet/ControlNet.cpp [hunk 1]
```diff
@@ -554,16 +602,7 @@ namespace Isis {
...
-            // If the edge already exists, this just returns it. (The use of a set
-            // forces the edges to be unique.)
-            ImageConnection connection;
-            bool edgeAdded;
-            boost::tie(connection, edgeAdded) = boost::add_edge(m_vertexMap[serial],
-                                                       m_vertexMap[sn],
-                                                       m_controlGraph);
-            m_controlGraph[connection].strength++;
-
+            bool edgeAdded = addEdge(serial, sn);
```
Comment (krlberry) [inline isis/src/control/objs/ControlNet/ControlNet.cpp:615]: For whatever reason, I thought doing it this way was clearer than:

```
if( addEdge(serial, sn) ){
  emit networkModified(GraphModified);
}
```
Let me know if you prefer this way, as I wasn't completely sure which would be better.

Comment (jessemapel) [reply isis/src/control/objs/ControlNet/ControlNet.cpp:615]: I think it's better to include the signals in the addEdges and removeEdge methods. That way we always notify when the graph changes.

**Review by jessemapel — CHANGES_REQUESTED** [review]

File: isis/src/control/objs/ControlNet/ControlNet.cpp (hunk 1 above)
Comment (krlberry) [reply isis/src/control/objs/ControlNet/ControlNet.cpp:615]: Agreed, but -- putting the signals in the methods changes the unit test output for `ControlNet` which... seems incorrect. I'll call you on hangouts about this later.

Comment (jessemapel) [reply isis/src/control/objs/ControlNet/ControlNet.cpp:615]: sounds good

**Review by jessemapel — APPROVED** [review]

**Pull request merged**